x/slashing/abci.go

```diff
@@ -56,6 +56,55 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 		if writeInfo == nil {
 			panic("Expected slashing write info to be non-nil")
 		}
+		// Check if we need to resize the array if there was recently a change in slashing window size
+		window := k.SignedBlocksWindow(ctx)
+		missedInfo, found := k.GetValidatorMissedBlocks(ctx, writeInfo.ConsAddr)
+		missedBlockLen := int64(len(missedInfo.MissedBlocks))
+		if found && window != missedBlockLen {
+			// we need to resize the missed block array AND update the signing info accordingly
+			switch {
+			case missedBlockLen < window:
+				// missed block array too short, lets expand it
```

Comment:
is the logic here, that if we're going from 5 -> 7 window:
- before: [0..5]
- after: [nil, nil, 0..5]
?

Reply:
the logic is such that our index offset should still point to the same value so that the next value to get overwritten is the oldest value